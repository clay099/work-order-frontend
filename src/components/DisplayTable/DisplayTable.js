import React from "react";
import { Table, Typography, Paper, TableContainer } from "@material-ui/core/";
import TableHeadings from "../TableHeadings/TableHeadings";
import TableContents from "../TableContents/TableContents";
import useStyles from "./styles";

const DisplayTable = ({
	projectData,
	headingList,
	tableTitle,
	tableType = "full",
}) => {
	const classes = useStyles();

	return (
		<div>
			<TableContainer component={Paper} className={classes.table}>
				<Typography
					component="h2"
					variant="h6"
					color="primary"
					gutterBottom
				>
					{tableTitle}
				</Typography>
				<Table size="small" dense table>
					<TableHeadings headings={headingList} />
					<TableContents
						bodyData={projectData}
						tableType={tableType}
					/>
				</Table>
			</TableContainer>
		</div>
	);
};

export default DisplayTable;
