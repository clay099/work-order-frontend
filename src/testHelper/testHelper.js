// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
