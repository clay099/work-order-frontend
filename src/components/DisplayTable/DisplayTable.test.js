import React from "react";
import DisplayTable from "./DisplayTable";
import { render, formDataState } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	projectData: [formDataState.formData],
	headingList: ["heading 1"],
	tableTitle: "title",
};

const initialState = {
	projects: { projectList: [formDataState.formData] },
};

describe("DisplayTable Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<DisplayTable {...params} />, {
			initialState,
		});
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<DisplayTable {...params} />, {
			initialState,
		});
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<DisplayTable {...params} />, {
			initialState,
		});
		const value = getByText("title");
		expect(value).toBeInTheDocument();
	});
});
