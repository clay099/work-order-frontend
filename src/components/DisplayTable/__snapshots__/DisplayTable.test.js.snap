// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayTable Component matches snapshot 1`] = `[Function]`;
